### PR TITLE
jupyter and ipython code-blocks

### PR DESCRIPTION
**Pull request by FHaase**
Add support for >>> blocks outside of code-blocks
Add ipython as identifier for python blocks

### CODE REVIEW

File: flake8_rst/rst.py [hunk 1]
```diff
@@ -3,13 +3,14 @@
 
 RST_RE = re.compile(
     r'(?P<before>'
-    r'^(?P<indent> *)\.\. (code-block|sourcecode):: (python|pycon)\n'
+    r'^(?P<indent> *)\.\. (code-block|sourcecode|ipython):: (python|pycon)\n'
```
Comment (kataev) [inline flake8_rst/rst.py]: If i read properly, ipython [directive](https://ipython.org/ipython-doc/rel-0.13.2/development/ipython_directive.html) use without (python|pycon)

And ipython console output contain square brackets, without `>>>`, how does it work?

Comment (FHaase) [reply flake8_rst/rst.py]: maybe I've been a little bit too enthusiastic, however in [pandas](https://github.com/pandas-dev/pandas/blob/master/doc/source/enhancingperf.rst) they use directives like
`.. ipython:: python` followed by plain python code. [with some roles like :suppress: sometimes]

These cases would be covered as well.

File: flake8_rst/rst.py [hunk 2]
```diff
@@ -18,31 +19,41 @@
...
+    r'^(((?P=indent)(>>>|...)(.*))?\n)+)'
+    r'))',
+    re.MULTILINE,
+)
+
 
 def find_sourcecode(src):
-    for match in RST_RE.finditer(src):
-        origin_code = match.group('code')
+    for expression in [RST_RE, ANCHOR_RE]:
```
[the remark this answers is not in the document]
Comment (FHaase) [reply flake8_rst/rst.py]: Great idea, much cleaner.

File: flake8_rst/rst.py (hunk 1 above)
Comment (kataev) [reply flake8_rst/rst.py]: okey

**Comment by kataev** [comment]
Thanks

**Pull request merged**